extension/src/popup/SorobanContext.tsx

```diff
@@ -0,0 +1,48 @@
+import React from "react";
+import { useSelector } from "react-redux";
+import { BASE_FEE, SorobanRpc, TransactionBuilder } from "stellar-sdk";
+
+import { settingsNetworkDetailsSelector } from "./ducks/settings";
+
+export const hasSorobanClient = (
+  context: SorobanContextInterface,
+): context is Required<SorobanContextInterface> =>
+  context.server !== undefined && context.newTxBuilder !== undefined;
+
+export interface SorobanContextInterface {
+  server: SorobanRpc.Server;
+  newTxBuilder: (fee?: string) => Promise<TransactionBuilder>;
+}
+
+export const SorobanContext = React.createContext(
+  {} as SorobanContextInterface,
+);
+
+export const SorobanProvider = ({
+  children,
+  pubKey,
+}: {
+  children: React.ReactNode;
+  pubKey: string;
+}) => {
+  const networkDetails = useSelector(settingsNetworkDetailsSelector);
+  const { sorobanRpcUrl = "" } = networkDetails;
```

Comment:
I simplified the logic here as before we were dealing with edge cases where network's didn't have an RPC. I think we can assume everyone will be including an RPC url going forward, but I'll test the fail case of this